include/kernel.h

```diff
@@ -376,7 +377,17 @@ struct __thread_entry {
 struct _thread_base {
 
 	/* this thread's entry in a ready/wait queue */
-	sys_dnode_t k_q_node;
+	union {
```

Comment:
IMHO usage of qnode_dump in outside ready queue context (mailbox.c and pipe.c) calls for better name for this field. What about:

```
union
{
sys_dlist_t k_q_node; // Generic queue node used to put the thread on the list.
struct rbnode k_q_node_rb; // Node for ready queue when CONFIG_WAITQ_FAST is used.
} 
```

Reply:
The direct queuing in mailbox and pipe is IMHO a pretty bad layering violation and needs to be removed in favor of some other mechanism (maybe extending the new priq abstraction a tiny bit to allow for simple list management).  IPC tools shouldn't be dealing with scheduler internals.  So I agree, but I'd rather fix it on the other side. :)

But FWIW: the "dumb" in that name isn't an informal or internal thing. It's the "dumb list" variant of thread queueing, and as it has scaling impact I thought it should be explicit.  But if it bothers you strongly I can rename it or otherwise hide the "dumb" (behind a more abstracted struct containing the union, maybe) from kernel.h